applications/templates/argo-auth.yaml

```diff
@@ -0,0 +1,35 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: argocd-config
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "3"
+  labels:
+    app: argocd-config
+spec:
+  template:
+    metadata:
+      name: argocd-config
+      labels:
+        app: argocd-config
+    spec:
+      serviceAccountName: argocd-server
+      containers:
+        - name: oc
+          image: "quay.io/openshift/origin-cli:latest"
+          imagePullPolicy: "IfNotPresent"
+          command:
+          - /bin/sh
+          - -c
+          - >
+              HOSTNAME={{ (lookup "route.openshift.io/v1" "Route" .Release.Namespace "argocd").spec.host }}
+              TOKEN={{ (lookup "v1" "Secret" .Release.Namespace "dex-auth-secret").data.token | b64dec }}
+              ADMINS={{ (lookup "v1" "Secret" .Release.Namespace "argocd-group-config").data.ADMIN_GROUPS | b64dec }}
+              READERS={{ (lookup "v1" "Secret" .Release.Namespace "argocd-group-config").data.READ_GROUPS | b64dec }}
+              API=https://kubernetes.default.svc:443;
+              oc annotate sa argocd-dex-server serviceaccounts.openshift.io/oauth-redirecturi.{{ .Release.Namespace }}=https://$HOSTNAME/api/dex/callback;
+              oc patch cm argocd-cm -p '{"data":{"url": "https://$HOSTNAME"}}';
+              oc patch cm argocd-cm -p '{"data":{"dex.config": "connectors:\n  - type: openshift\n    id: openshift\n    name: OpenShift\n    config:\n      issuer: $API\n      clientID: system:serviceaccount:{{ .Release.Namespace }}:argocd-dex-server\n      clientSecret: $TOKEN\n      redirectURI: https://$HOSTNAME/api/dex/callback\n      insecureCA: true\n      groups:\n{{ range (lookup "v1" "Secret" .Release.Namespace "argocd-group-config").data.ADMIN_GROUPS | b64dec | split "," }}        - {{ . }}\n{{- end }}{{ range (lookup "v1" "Secret" .Release.Namespace "argocd-group-config").data.READ_GROUPS | b64dec | split "," }}        - {{ . }}\n{{- end }}"}}';
+              oc patch cm argocd-rbac-cm -p "{\"data\":{\"policy.csv\": \"{{- range (lookup "v1" "Secret" .Release.Namespace "argocd-group-config").data.ADMIN_GROUPS | b64dec | split "," }}g, {{ . }}, role:admin\n{{- end }}{{- range (lookup "v1" "Secret" .Release.Namespace "argocd-group-config").data.READ_GROUPS | b64dec | split "," }}g, {{ . }}, role:readonly\n{{- end }}\"}}"
+      restartPolicy: OnFailure
```

Comment:
I think I understand the gist of this but can't say I fully grok everything it does. It seems to be dynamic in how it looks things up which should address my concern that something is hard coded